cli/cmd/install_helm_test.go

```diff
@@ -0,0 +1,207 @@
+      "CrtExpiry":"Jul 30 17:21:14 2020",
+      "CrtExpiryAnnotation":"%s",
+      "TLS":{
+        "KeyPEM":"test-key-pem",
+        "CrtPEM":"test-crt-pem"
+      }
+    }
+  },
+  "Configs": null,
+  "Proxy":{
+    "Image":{
+      "Version":"test-proxy-version"
+    }
+  },
+  "ProxyInit":{
+    "Image":{
+      "Version":"test-proxy-init-version"
+    }
+  },
+  "ProxyInjector":{
+    "KeyPEM":"test-proxy-injector-key-pem",
+    "CrtPEM":"test-proxy-injector-crt-pem"
+  },
+  "ProfileValidator":{
+    "KeyPEM":"test-profile-validator-key-pem",
+    "CrtPEM":"test-profile-validator-crt-pem"
+  },
+  "Tap":{
+    "KeyPEM":"test-tap-key-pem",
+    "CrtPEM":"test-tap-crt-pem"
+  }
+}`
+	overrideConfig := &pb.Config{
+		Raw: fmt.Sprintf(overrideJSON, k8s.IdentityIssuerExpiryAnnotation),
+	}
+
+	rendered, err := renderutil.Render(chartControlPlane, overrideConfig, releaseOptions)
+	if err != nil {
+		t.Fatal("Unexpected error", err)
+	}
+
+	var buf bytes.Buffer
+	for _, template := range chartControlPlane.Templates {
+		source := chartName + "/" + template.Name
+		v, exists := rendered[source]
+		if !exists {
+			// skip partial templates
+			continue
+		}
+		buf.WriteString("---\n# Source: " + source + "\n")
+		buf.WriteString(v)
+	}
+
+	// pin the uuid in the linkerd-config config map
+	re := regexp.MustCompile(`"uuid":".*"`)
+	result := re.ReplaceAllString(buf.String(), `"uuid":"test-install-uuid"`)
+
+	diffTestdata(t, goldenFileName, result)
+}
+
+func chartControlPlane(t *testing.T, ha bool) *pb.Chart {
+	values, err := readTestValues(t, ha)
+	if err != nil {
+		t.Fatal("Unexpected error", err)
+	}
+
+	chartPartials := chartPartials(t)
+
+	chart := &pb.Chart{
+		Metadata: &pb.Metadata{
+			Name: helmDefaultChartName,
+			Sources: []string{
+				filepath.Join("..", "..", "..", "charts", "linkerd2"),
+			},
+		},
+		Templates: []*pb.Template{
+			{Name: "templates/namespace.yaml"},
+			{Name: "templates/identity-rbac.yaml"},
+			{Name: "templates/controller-rbac.yaml"},
+			{Name: "templates/heartbeat-rbac.yaml"},
+			{Name: "templates/web-rbac.yaml"},
+			{Name: "templates/serviceprofile-crd.yaml"},
+			{Name: "templates/trafficsplit-crd.yaml"},
+			{Name: "templates/prometheus-rbac.yaml"},
+			{Name: "templates/grafana-rbac.yaml"},
+			{Name: "templates/proxy-injector-rbac.yaml"},
+			{Name: "templates/sp-validator-rbac.yaml"},
+			{Name: "templates/tap-rbac.yaml"},
+			{Name: "templates/psp.yaml"},
+			{Name: "templates/_validate.tpl"},
+			{Name: "templates/_affinity.tpl"},
+			{Name: "templates/_config.tpl"},
+			{Name: "templates/_helpers.tpl"},
+			{Name: "templates/config.yaml"},
+			{Name: "templates/identity.yaml"},
+			{Name: "templates/controller.yaml"},
+			{Name: "templates/heartbeat.yaml"},
+			{Name: "templates/web.yaml"},
+			{Name: "templates/prometheus.yaml"},
+			{Name: "templates/grafana.yaml"},
+			{Name: "templates/proxy-injector.yaml"},
+			{Name: "templates/sp-validator.yaml"},
+			{Name: "templates/tap.yaml"},
```

Comment:
What do you think about extracting this list from here and from `render` in `install.go` and putting it in some common place under the `cmd` package? Would of course require some helper function to account for the different sets of templates those two callers require, and also the different types (`pb.Template` vs (`chartutil.BufferedFile`)

Reply:
Good idea. Will adding the filepath to a string slice work for now, and let the callers decide what types they need?

At first, I was hoping to move all these templates references to `charts.Chart`, as it makes sense that a `Chart` instance knows where to find its templates. But then, that will require more code change in `Chart` to support multi-stage templates and some of the `inject` code too. So I decided not to go down that route for now.